packages/connect-core/src/utils/path/calculatePath.ts

```diff
@@ -181,34 +150,26 @@ async function calculateForwardingPath(
  */
 export async function calculateTransactionPath(
   sender: string,
-  destination: string,
+  destinationApp: App,
   methodSignature: string,
   params: any[],
   apps: App[],
   provider: ethers.providers.Provider,
   finalForwarder?: string //Address of the final forwarder that can perfom the action. Needed for actions that aren't in the ACL but whose execution depends on other factors
-): Promise<PathData> {
-  // Get the destination app
-  const destinationApp = apps.find((app) => app.address == destination)
-  if (!destinationApp) {
-    throw new Error(
-      `Transaction path destination (${destination}) is not an installed app`
-    )
-  }
-
-  // Make sure the method signature is correct
-  const method = validateMethod(destination, methodSignature, destinationApp)
-
-  const finalForwarderProvided = finalForwarder
-    ? ethers.utils.isAddress(finalForwarder)
-    : false
+): Promise<Transaction[]> {
+  // The direct transaction we eventually want to perform
   const directTransaction = await createDirectTransactionForApp(
     sender,
     destinationApp,
-    method.sig,
+    methodSignature,
     params
   )
 
+  const finalForwarderProvided = finalForwarder
+    ? ethers.utils.isAddress(finalForwarder)
+    : false
+
+  const method = getAppMethod(destinationApp, methodSignature)
```

Comment:
We are only using the `method` to know if it's protected by a role. I wonder if we can get this information in other way 🤔